lib/plugin.js

```diff
@@ -0,0 +1,68 @@
+const RabbitMQ = require('./rabbitmq')
+const { request } = require('undici')
+
+/** @param {import('fastify').FastifyInstance} app */
+module.exports = async function (app, options) {
+  app.log.info('Starting rabbitmq-hooks plugin')
+  const { rabbitmq } = app.platformatic.config
+
+  const { url, generateExchange, exchanges } = rabbitmq
+
+  const rabbit = new RabbitMQ({ logger: app.log, generateExchange })
+  await rabbit.connect(url)
+
+  // create a listener for each exchange
+  for (const exchange of exchanges) {
+    const { name, routingKey, targetUrl } = exchange
+
+    app.log.info(`Listening to exchange ${name}`)
+    if (!routingKey) {
+      app.log.info('No routing key specified')
+    } else {
+      app.log.info(`Routing key: ${routingKey}`)
+    }
+    await rabbit.listen(name, routingKey, async (msg) => {
+      const message = msg?.content?.toString()
+      app.log.debug('Received', message)
+      try {
+        await request(targetUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
```

Comment:
Content type should be json unless it's specified inside the headers map (or the headers map as a json content type as default.